Downloader.m

```diff
@@ -118,4 +126,19 @@ - (void)stopDownload
   return _params.errorCallback(error);
 }
 
+
+- (void)URLSessionDidFinishEventsForBackgroundURLSession:(NSURLSession *)session
+{
+    NSLog(@"URLSessionDidFinishEventsForBackgroundURLSession");
+
+    AppDelegate *appDelegate = (AppDelegate *)[[UIApplication sharedApplication] delegate];
+
+    
+    if (appDelegate.sessionCompletionHandler) {
```

Comment:
So this is a stateful background delegation?  Something else in the bridge code sets that callback when we start to process something in the background?